keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
-    // resolves to the corresponding value of the instance under evaluation.
-    // For example:
+    // In a cluster with multiple scheduler instances, two pods might get
+    // scheduled concurrently by different schedulers. When they reference
+    // the same ResourceClaim which may only get used by one pod at a time,
+    // only one pod can be scheduled.
     //
-    //    attributes.quantity["a"].isGreaterThan(quantity("0")) &&
-    //    attributes.stringslice["b"].isSorted()
-    Selector string
+    // Both schedulers try to add their pod to the claim.status.reservedFor
+    // field, but only the update that reaches the API server first gets
+    // stored. The other one fails with a conflict error and the scheduler
+    // which issued it knows that it must put the pod back into the queue,
+    // waiting for the ResourceClaim to become usable again.
+    //
+    // There can be at most 32 such reservations. This may get increased in
+    // the future, but not reduced.
+    //
+    // +listType=map
+    // +listMapKey=uid
+    // +patchStrategy=merge
+    // +patchMergeKey=uid
+    // +optional
+    ReservedFor []ResourceClaimConsumerReference
 }
 ```
 
-NamedResourcesFilter and NamedResourcesRequest currently have the same
-content. Despite that, they are defined as separate structs because that might
-change in the future.
-
-
 ##### Allocation result
 
 ```go
 // AllocationResult contains attributes of an allocated resource.
 type AllocationResult struct {
-    // ResourceHandles contain the state associated with an allocation that
-    // should be maintained throughout the lifetime of a claim. Each
-    // ResourceHandle contains data that should be passed to a specific kubelet
-    // plugin once it lands on a node.
+    // ControllerName is the name of the DRA driver which handled the
+    // allocation. That driver is also responsible for deallocating the
+    // claim. It is empty when the claim can be deallocated without
+    // involving a driver.
     //
-    // Setting this field is optional. It has a maximum size of 32 entries.
-    // If null (or empty), it is assumed this allocation will be processed by a
-    // single kubelet plugin with no ResourceHandle data attached. The name of
-    // the kubelet plugin invoked will match the DriverName set in the
-    // ResourceClaimStatus this AllocationResult is embedded in.
+    // A driver may allocate devices provided by other drivers, so this
+    // driver name here can be different from the driver names listed in
+    // the results.
+    //
+    // This is an alpha field and requires enabling the DRAControlPlaneController
+    // feature gate.
+    //
+    // +optional
+    ControllerName string
+
+    // This field holds configuration for all drivers which
+    // satisfied requests in this claim. The configuration applies to
+    // the entire claim.
+    //
+    // +optional
+    // +listType=atomic
+    Config []ConfigurationParameters
```

Comment:
I would put this as the last object in the struct

Reply:
Done.